doc(self-managed): improve irsa troubleshooting for existing setups (#7151)
camunda/camunda-docs: versioned_docs/version-8.7/self-managed/setup/deploy/amazon/amazon-eks/irsa.md

[Instance Metadata Service](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) is a default fallback for the AWS SDK due to the [default credentials provider chain](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials-chain.html). Within the context of Amazon EKS, it means a pod will automatically assume the role of a node. This can hide many problems, including whether IRSA was set up correctly or not, since it will fall back to IMDS in case of failure and hide the actual error.

Thus, if nothing within your cluster relies on the implicit node role, we recommend disabling it by configuring the `http_put_response_hop_limit` to 1. This decreases the default value of two to one, which results in pods not being allowed to assume the role of the node anymore.

### Configuring IMDS hop limit

> **For new node groups**, using a Terraform module like the [Amazon EKS module](https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest), you can define the following:

Reviewer: ```suggestion
**For new node groups** that use a Terraform module such as the [Amazon EKS module](https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest), you can define the following:
```

}
```

:::note Enabled by default in the terraform reference architecture of EKS

In the [reference architecture with terraform](terraform-setup.md), this setting is configured like that by default.

:::

**For existing worker node instances**, you can modify the hop limit using the AWS CLI:

```bash
aws ec2 modify-instance-metadata-options \
    --instance-id <instance-id> \
    --http-put-response-hop-limit 1
```

> Replace `<instance-id>` with your actual EC2 instance ID. You'll need to run this command for each worker node in your cluster.

Reviewer: ```suggestion
Replace `<instance-id>` with your actual EC2 instance ID. You must run this command for each worker node in your cluster.
```
